staging/src/k8s.io/client-go/tools/cache/thread_safe_store.go

```diff
@@ -48,6 +50,7 @@ type ThreadSafeStore interface {
 	Replace(map[string]interface{}, string)
 	Index(indexName string, obj interface{}) ([]interface{}, error)
 	IndexKeys(indexName, indexedValue string) ([]string, error)
+	RV() string
```

Comment:
This doesn't update when deleting an object or when a bookmark is observed ... are those important?

What are we intending to use this for? The linked metrics example compares against latest list version which would grow for deletes and bookmarks, and would apparently make this diverge even if no relevant object updates for this controller were being received as add/update/replaces that would make this grow

Reply:
At least for the metrics, we will only check when the resource version grows so there's no risk of divergence. However with deletion you raise a good point, but with the way we currently delete there is no way for us to get the resource version, we probably either want to pass in the deleted object if it exists or something along those line. We might also want to do this on top of the work done in https://github.com/kubernetes/kubernetes/pull/134937 which adds the bookmark to the store. At least for us though, the use is just to get the resource version observed by the cache during periods of churn.

Reply:
I think we can also make use of resourceVersion info in the delete, but that means a peer `DeleteWithObject(key string, obj interface{})` method in the store